errors.go

```diff
@@ -0,0 +1,90 @@
+package hedera
+
+import (
+	"fmt"
+	"reflect"
+)
+
+type ErrMaxQueryPaymentExceeded struct {
+	// The cost of the query that was attempted as returned by QueryBuilder.Cost
+	QueryCost Hbar
+	// The limit for a single automatic query payment, set by
+	// Client.SetMaxQueryPayment(int64) or QueryBuilder.SetMaxQueryPayment(uint64).
+	MaxQueryPayment Hbar
+	// Name of the query builder class used for output
+	query string
+}
+
+func newErrorMaxQueryPaymentExceeded(builder *QueryBuilder, queryCost Hbar, maxQueryPayment Hbar) ErrMaxQueryPaymentExceeded {
```

Comment:
Max query payment shouldn't need to now about the query that was being run.

Reply:
This is how the Java SDK handles it, I'll leave it as is for now